common/src/main/java/io/netty5/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -37,11 +38,17 @@ public abstract class AbstractScheduledEventExecutor extends AbstractEventExecut
     private static final RunnableScheduledFutureNode<?>[]
             EMPTY_RUNNABLE_SCHEDULED_FUTURE_NODES = new RunnableScheduledFutureNode<?>[0];
 
-    private PriorityQueue<RunnableScheduledFutureNode<?>> scheduledTaskQueue;
+    private ScheduledTaskQueue<RunnableScheduledFutureNode<?>> scheduledTaskQueue;
+    private Supplier<ScheduledTaskQueue<RunnableScheduledFutureNode<?>>> scheduledTaskQueueSupplier;
 
     protected AbstractScheduledEventExecutor() {
     }
 
+    protected AbstractScheduledEventExecutor(
+            Supplier<ScheduledTaskQueue<RunnableScheduledFutureNode<?>>> scheduledTaskQueueSupplier) {
+        this.scheduledTaskQueueSupplier = requireNonNull(scheduledTaskQueueSupplier, "scheduledTaskQueueSupplier");
+    }
```

Comment:
### Discussion
```java
Supplier<ScheduledTaskQueue> taskQueueSupplier = () - new UserCustomScheduledQueue or HashedWheelTimer();

// call super(taskQueueSupplier) -> we can pass taskQueueSupplier to AbtractSceduledEventExecutor.
SingleThreadEventLoop eventLoop = new SingleThreadEventLoop(taskQueueSupplier);

// call new SingleThreadEventLoop(taskQueueSupplier) multiple times internally
MultiThreadsEventLoopGroup eventLoops = new MultiThreadsEventLoopGroup(taskQueueSupplier);
```
Q1. Cause scheduledTaskQueue instance is created by on-demand (see `scheduledTaskQueue() L69`), I think we can pass `Supplier<ScheduledTaskQueue>` to AbractScheduledEventExecutor's constructor to create our custom scheduledTaskQueue on-demand too

Also, On MultiThreadEventLoopGroup, it creates multiple SingleThreadEventLoop. so to set different scheduledTaskQueue on each SingleThreadEventLoop, we need `Supplier<ScheduledTaskQueue>`.



``` java 
    public SingleThreadEventLoop(ThreadFactory threadFactory, IoHandler ioHandler,
                                 Supplier<ScheduledTaskQueue> supplier) { // new!
        super(supplier);
        ...
    }

    public MultithreadEventLoopGroup(Executor executor,
                                     IoHandlerFactory ioHandlerFactory,
                                     Supplier<ScheduledTaskQueue> supplier) { // new!
        ....
    }

    protected EventLoop newChild(Executor executor, .. , Supplier<ScheduledTaskQueue> supplier)  {
        return new SingleThreadEventLoop(executor, .. , supplier); // pass supplier to each event loop!
    }

```
Q2. To pass our `Supplier<ScheduledTaskQueue>` to AbractScheduledEventExecutor's constructor, I think we need additional constructor on SingleThreadEventLoop and MultiThreadsEventLoopGroup.

Reply:
Hello @He-Pin ! this my first PR on netty, so I make draft PR and want to check my rough direction to solve this issue is correct.

Can you simply check above my question? 🙇
It my direction is right, I'll push more detailed code&test~!